lib/tenstorrent/bh_arc/gddr.c

```diff
@@ -303,6 +337,9 @@ static int InitMrisc(void)
 		}
 	}
 
+	k_free(buf);
+	buf = NULL;
+
 	return 0;
```

Comment:
```suggestion
```suggestion
	rc = 0;

free_buff:
	k_free(buf);

	return rc;
``````